### DISCUSSION TITLE
page extract with pdf4llm.to_markdown not extracting the first line of the page in specific scenarios

### DISCUSSION THREAD

**Discussion by leelaraj72** [pymupdf/pymupdf4llm#196] [Q&A] [answered]
I am using following simple python code to extract a page (and the corresponding text) from a pdf file
==
import pdf4llm
import pprint

data = pdf4llm.to_markdown("../docs/abc.pdf", pages=[13], page_chunks=True)
pprint.pprint(data[0])
==

The first line is not extracted always if the first line is a plain text. For Ex: if the page has following text
==
promotional modifiers where a service item gets added automatically when a certain
finished good is added to the order. This helps in cutting down order creation time, adds 
efficiency and accuracy of order creator, and enables companies to implement service
...
==
then data[0]['text'] has 
==
'finished good is added to the order. This helps in cutting down order '
 'creation time, adds\n'
 'efficiency and accuracy of order creator, and enables companies to implement '
 'service\n'
...
==

Did anyone else face the same problem?

**Comment by JorjMcKie** [comment, marked as answer]
Try setting the margin parameter. Default is `margins=(0, 50, 0, 50)` which ignores stripes of height 50 at top and bottom.
Using `margins=0` looks at the full page.

**Reply by Fianax** [reply]
I've got the margins under control.
That's what I call the method:

```
md_text_list_data_page = pdf4llm.to_markdown(
            doc = document, 
            margins = 0,
            page_chunks = True,
            hdr_info = my_custom_section
        )
```

**Reply by Fianax** [reply]
do I set the margins to 0 but not at all, or do I have to set them to (0, 0, 0, 0, 0)?

**Reply by JorjMcKie** [reply]
`margins=0` will auto-set all 4 borders to that value.

**Reply by JorjMcKie** [reply]
Please look at the documentation: either a single value, a pair of values or 4-tuples can be specified.

**Reply by Fianax** [reply]
then the problem is not the margins, but something else that makes it not return all the text in the ‘span’.

**Reply by JorjMcKie** [reply]
Checkout the page's spans and see what is happening:
```python
blocks = page.get_text("dict", flags=TEXTFLAGS_TEXT)["blocks"]
spans = [s for b in blocks for l in b["lines"] for s in l["spans"]]
for s in spans:
    print(f"{s['text']=}"
```

**Reply by Fianax** [reply]
With this code snippet it does return all the lines of code from the example pdf.

Although with the code that you have passed me it works for what I wanted (thanks for your code 💯 ) but the hdr_info would still not give all the text in the ‘span’ value of the method that is passed to it.

**Reply by Fianax** [reply]
I have also tried to use the flag=TEXTFLAGS_TEXT but I have not been able to import it from anywhere(fitz.).

What numerical value does it have to put it?

**Reply by JorjMcKie** [reply]
`pymupdf.TEXTFLAGS_TEXT` of course!!!

**Reply by JorjMcKie** [reply]
The check in hdr_info will only called for the **_first span of a line_**  - not for every span.

**Reply by Fianax** [reply]
oki, thanks for everything
